HelpSource/Classes/Object.schelp

```diff
@@ -204,6 +208,313 @@ a === b; // not identical
 "worth trying" == "worth trying"; // equal
 ::
 
+Discussion::
+
+list::
+## link::Classes/SimpleNumber:: and its subclasses link::Classes/Float:: and link::Classes/Integer::: anchor::equality & identity discussion::
+
+code::==:: is overridden by link::Classes/SimpleNumber:: to check for equality.
+For example, 5.0 and 5 are equal in value, but 5.0 is an instance of Float and 5 is an instance of Integer. So the equality and identity being tested are different.
+list::
+
+## Equality:
+
+code::
+5.0 == 5 // equal
+::
+
+teletype::
+-> true
+::
+
+If they are equal, then their emphasis::hash table indices:: are equal and identical, i.e. the same number will always result in the same emphasis::hash index::.
+
+code::
+// Evaluate the following codes more than once and compare the results.
+5.hash
+5.0.hash
+::
+code::
+5.hash ==  5.0.hash
+5.hash === 5.0.hash
+::
+
+note::
+SuperCollider uses emphasis::64-bit double precision floats:: on the client (sclang) and emphasis::32-bit floats:: on the server (scsynth and supernova). The human-readable decimal representation of the float value may not be accurate. To check the exact representation of float in sclang, the code::.dump:: method (see: link::#-dump::) can be used.
```

Comment:
>This belongs in documentation for Float,

done! moved to SimpleNumber.schelp

>it also isn't true for a 32-bit build.
Thanks! corrected! 

**line 476 and 477**
This is due to the imprecise decimal representation of float
because the 64-bit SuperCollider uses emphasis::64-bit double precision floats:: on the client (sclang) and emphasis::32-bit floats:: on the server (scsynth and supernova).